opensearch/catalog-opensearch-endpoint/src/main/java/org/codice/ddf/endpoints/ResourceUriTransformerCatalogFrameworkProxy.java

```diff
@@ -0,0 +1,199 @@
+        ResourceNotFoundException, ResourceNotSupportedException {
+        return catalog.getEnterpriseResource(request);
+    }
+
+    public Map<String, Set<String>> getEnterpriseResourceOptions(String metacardId)
+        throws ResourceNotFoundException {
+        return catalog.getEnterpriseResourceOptions(metacardId);
+    }
+
+    public ResourceResponse getLocalResource(ResourceRequest request) throws IOException,
+        ResourceNotFoundException, ResourceNotSupportedException {
+        return catalog.getLocalResource(request);
+    }
+
+    public Map<String, Set<String>> getLocalResourceOptions(String metacardId)
+        throws ResourceNotFoundException {
+        return catalog.getLocalResourceOptions(metacardId);
+    }
+
+    public ResourceResponse getResource(ResourceRequest request, String resourceSiteName)
+        throws IOException, ResourceNotFoundException, ResourceNotSupportedException {
+        return catalog.getResource(request, resourceSiteName);
+    }
+
+    public Map<String, Set<String>> getResourceOptions(String metacardId, String sourceId)
+        throws ResourceNotFoundException {
+        return catalog.getResourceOptions(metacardId, sourceId);
+    }
+
+    public Set<String> getSourceIds() {
+        return catalog.getSourceIds();
+    }
+
+    public SourceInfoResponse getSourceInfo(SourceInfoRequest sourceInfoRequest)
+        throws SourceUnavailableException {
+        return catalog.getSourceInfo(sourceInfoRequest);
+    }
+
+    public QueryResponse query(QueryRequest query) throws UnsupportedQueryException,
+        SourceUnavailableException, FederationException {
+        return catalog.query(query);
+    }
+
+    public QueryResponse query(QueryRequest queryRequest, FederationStrategy strategy)
+        throws SourceUnavailableException, UnsupportedQueryException, FederationException {
+        return catalog.query(queryRequest, strategy);
+    }
+
+    public BinaryContent transform(Metacard metacard, String transformerId,
+            Map<String, Serializable> requestProperties) throws CatalogTransformerException {
+        return catalog.transform(metacard, transformerId, requestProperties);
+    }
+
+    /**
+     * Converts the resource URIs found in the {@link SourceResponse} {@link Metacard}s using the
+     * {@link ActionProvider} injected in the constructor before delegating the call to the proxied
+     * {@link CatalogFramework} object.
+     */
+    public BinaryContent transform(SourceResponse response, String transformerId,
+            Map<String, Serializable> requestProperties) throws CatalogTransformerException {
+        transform(response);
+        return catalog.transform(response, transformerId, requestProperties);
+    }
+
+    public UpdateResponse update(UpdateRequest updateRequest) throws IngestException,
+        SourceUnavailableException {
+        return catalog.update(updateRequest);
+    }
+
+    private void transform(SourceResponse queryResponse) {
```

Comment:
Some comments to your thoughts:
1. The client code doesn't have to be updated, only the transformer code does. Additionally, maybe formats support multiple resource urls so this could be just adding a new url to the format. With the proxy implementation you're still having to 'change' the clients by modifying their dependency injection to include the proxy.
2. I would prefer having a talk about something as simple as deciding a new attribute name rather than updating all current endpoints and requiring all future endpoints to embed / use a proxy catalogframework in order to get something as simple as an external url.